setup.py

```diff
@@ -174,15 +175,20 @@ def finalize_options(self):
 
     def run(self):
         import os
-        os.system("find . -name '*.pyc' | xargs rm -f")
-        os.system("rm -f python-build-stamp-2.4")
-        os.system("rm -f MANIFEST")
-        os.system("rm -rf build")
-        os.system("rm -rf dist")
-        os.system("rm -rf doc/_build")
-        os.system("rm -f sample.tex")
-
-
+        
+        for root, dirs, files in os.walk(os.path.dirname(os.path.realpath(__file__))):
+            for file in files:
+                if file.endswith('.pyc') and os.path.isfile:
+                    os.remove(os.path.join(root, file))
+
+        if os.path.isfile("python-build-stamp-2.4"): os.remove("python-build-stamp-2.4")
```

Comment:
We are safe here. The recursive delete is only for finding and taking out the `.pyc` files that too the ones which reside in the dir having `setup.py` and below.

Moreover, I "`os.chdir()`"ed to the dir having `setup.py` before deleting the other files just to be safe and then came back to the original working dir.

No way things could go wrong here.